File: hospital.client/src/types/AppointmentResponse.ts
import type { AppointmentStatusResponse } from "./AppointmentStatusResponse";
import type { BranchResponse } from "./BranchResponse";
import type { SpecialtyResponse } from "./SpecialtyResponse";
import type { UserResponse } from "./UserResponse";

export interface AppointmentResponse {
  id: number;
  patientId: number;
  doctorId?: number | null;
  specialtyId: number;
  branchId: number;
  appointmentStatusId: number;
  appointmentDate: string;
  reason: string;
  amount: number;
  priority: number;
  arrivalTime?: string | null;
  notes?: string | null;
  followUpType?: number | null;
  parentConsultationId?: number | null;
  state: number;
  createdAt: string;
  createdBy: number;
  updatedBy?: number | null;
  updatedAt?: string | null;

  specialty?: SpecialtyResponse | null;
  branch?: BranchResponse | null;
  appointmentStatus?: AppointmentStatusResponse | null;
  patient?: UserResponse | null;
  doctor?: UserResponse | null;
}

export interface AppointmentRequest {
  id?: number | null;
  patientId?: number | null;
  doctorId?: number | null;
  specialtyId?: number | null;
  branchId?: number | null;
  appointmentStatusId?: number | null;
  appointmentDate?: string | null;
  reason?: string | null;
  amount?: number | null;
  priority?: number | null;
  arrivalTime?: string | null;
  notes?: string | null;
  followUpType?: number | null;
  parentConsultationId?: number | null;
  state?: number | null;
  createdBy?: number | null;
  updatedBy?: number | null;
}

File: Hospital.Server/Entities/Response/AppointmentResponse.cs
namespace Hospital.Server.Entities.Response
{
    public class AppointmentResponse
    {
        public long Id { get; set; }
        public long PatientId { get; set; }
        public long? DoctorId { get; set; }
        public long SpecialtyId { get; set; }
        public long BranchId { get; set; }
        public long AppointmentStatusId { get; set; }
        public string AppointmentDate { get; set; } = string.Empty;
        public string Reason { get; set; } = string.Empty;
        public decimal Amount { get; set; }
        public int Priority { get; set; }
        public string? ArrivalTime { get; set; }
        public string? Notes { get; set; }
        public int? FollowUpType { get; set; }
        public long? ParentConsultationId { get; set; }
        public int State { get; set; }
        public string CreatedAt { get; set; } = string.Empty;
        public long CreatedBy { get; set; }
        public long? UpdatedBy { get; set; }
        public string? UpdatedAt { get; set; }

        // Navigation responses
        public virtual SpecialtyResponse? Specialty { get; set; }
        public virtual BranchResponse? Branch { get; set; }
        public virtual AppointmentStatusResponse? AppointmentStatus { get; set; }
    }
}
